pkg/resources/istioingress/meshgateway.go

```diff
@@ -30,45 +30,57 @@ import (
 	kafkautils "github.com/banzaicloud/kafka-operator/pkg/util/kafka"
 )
 
-func (r *Reconciler) meshgateway(log logr.Logger, externalListenerConfig v1beta1.ExternalListenerConfig) runtime.Object {
+func (r *Reconciler) meshgateway(log logr.Logger, externalListenerConfig v1beta1.ExternalListenerConfig,
+	ingressConfig v1beta1.IngressConfig, ingressConfigName, defaultIngressConfigName string) runtime.Object {
+
 	mgateway := &istioOperatorApi.MeshGateway{
 		ObjectMeta: templates.ObjectMeta(
 			fmt.Sprintf(istioingressutils.MeshGatewayNameTemplate, externalListenerConfig.Name, r.KafkaCluster.Name),
-			labelsForIstioIngress(r.KafkaCluster.Name, externalListenerConfig.Name), r.KafkaCluster),
+			labelsForIstioIngress(r.KafkaCluster.Name, annotationName), r.KafkaCluster),
 		Spec: istioOperatorApi.MeshGatewaySpec{
 			MeshGatewayConfiguration: istioOperatorApi.MeshGatewayConfiguration{
-				Labels:             labelsForIstioIngress(r.KafkaCluster.Name, externalListenerConfig.Name),
-				ServiceAnnotations: externalListenerConfig.GetServiceAnnotations(),
+				Labels:             labelsForIstioIngress(r.KafkaCluster.Name, annotationName),
+				ServiceAnnotations: ingressConfig.GetServiceAnnotations(),
```

Comment:
Why `annotationName` is passed into `labelsForIstioIngress` instead of `externalListenerConfig.Name`?